contracts/modules/SwapsExternal.sol

```diff
@@ -92,10 +93,12 @@ contract SwapsExternal is VaultController, SwapsUser, ModuleCommonFunctionalitie
 
 			uint256 balanceBefore = sourceTokenContract.balanceOf(address(this));
 
-			IERC20(sourceToken).safeTransferFrom(msg.sender, address(this), sourceTokenAmount);
+			if (address(this) != msg.sender) {
+				IERC20(sourceToken).safeTransferFrom(msg.sender, address(this), sourceTokenAmount);
 
-			// explicit balance check so that we can support deflationary tokens
-			sourceTokenAmount = sourceTokenContract.balanceOf(address(this)).sub(balanceBefore);
+				// explicit balance check so that we can support deflationary tokens
+				sourceTokenAmount = sourceTokenContract.balanceOf(address(this)).sub(balanceBefore);
```

Comment:
- please dont support them =)
- if you have you might want to sync first (before token transfer), otherwise the contract's balance change would be applied as part of user's operation (it might increase or decrease user's  sourceTokenAmount)